server/src/test/java/org/elasticsearch/index/mapper/NumberFieldMapperTests.java

```diff
@@ -226,5 +254,33 @@ protected final Object generateRandomInputValue(MappedFieldType ft) {
         return randomBoolean() ? n : n.toString();
     }
 
+    @Override
+    @SuppressWarnings("unchecked")
+    protected <T> T compileScript(Script script, ScriptContext<T> context) {
+        if (context == LongFieldScript.CONTEXT) {
+            return (T) LongScriptFieldType.PARSE_FROM_SOURCE;
+        }
+        if (context == DoubleFieldScript.CONTEXT) {
+            return (T) DoubleScriptFieldType.PARSE_FROM_SOURCE;
```

Comment:
what is the effect of using the parse from source script here?

Reply:
It's just a convenient already-built script that we can use to ensure that compilation etc are happening correctly.